lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -469,7 +469,7 @@ public TopDocs searchAfter(ScoreDoc after, Query query, int numHits) throws IOEx
 
       @Override
       public TopScoreDocCollector newCollector() throws IOException {
-        return TopScoreDocCollector.create(cappedNumHits, after, TOTAL_HITS_THRESHOLD);
+        return TopScoreDocCollector.create(cappedNumHits, after, new GlobalHitsThresholdChecker(TOTAL_HITS_THRESHOLD));
```

Comment:
It should be created once and shared within the collectors ? We also don't need to use the `GlobalHitsThresholdChecker` if the executor is null or if there is a single slice.